src/components/calculator.js

```diff
@@ -1,32 +1,52 @@
+import calculate from '../Logic/calculate';
+import './styles.css';
+
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  onClickHandler = (e) => {
+    const data = e.target.innerHTML;
+    const result = calculate(this.state, data);
+    this.setState(result);
+  };
 
-class Calculator extends React.PureComponent {
   render() {
+    const { total, next, operation } = this.state;
     return (
       <div className="container">
-        <form>
-          <input type="text" />
-        </form>
+        <div className="input">
+          {total}
+          {operation}
+          {next}
+        </div>
         <div className="keypad">
-          <button type="button" name="A/C">A/C</button>
-          <button type="button" name="+/-">+/-</button>
-          <button type="button" name="%">%</button>
-          <button type="button" id="divide" name="/">&divide;</button>
-          <button type="button" name="7">7</button>
-          <button type="button" name="8">8</button>
-          <button type="button" name="9">9</button>
-          <button type="button" id="mult" name="*">&ndash;</button>
-          <button type="button" name="4">4</button>
-          <button type="button" name="5">5</button>
-          <button type="button" name="6">6</button>
-          <button type="button" id="less" name="-">&times;</button>
-          <button type="button" name="1">1</button>
-          <button type="button" name="2">2</button>
-          <button type="button" name="3">3</button>
-          <button type="button" id="plus" name="+">+</button>
-          <button type="button" name=".">.</button>
-          <button type="button" id="equal" name="=">=</button>
-          <button type="button" id="cero" name="0">0</button>
+          <button onClick={this.onClickHandler} type="button" name="A/C">AC</button>
+          <button onClick={this.onClickHandler} type="button" name="+/-">+/-</button>
+          <button onClick={this.onClickHandler} type="button" name="%">%</button>
+          <button onClick={this.onClickHandler} type="button" id="divide" name="÷">÷</button>
+          <button onClick={this.onClickHandler} type="button" name="7">7</button>
+          <button onClick={this.onClickHandler} type="button" name="8">8</button>
+          <button onClick={this.onClickHandler} type="button" name="9">9</button>
+          <button onClick={this.onClickHandler} type="button" id="mult" name="*">x</button>
+          <button onClick={this.onClickHandler} type="button" name="4">4</button>
+          <button onClick={this.onClickHandler} type="button" name="5">5</button>
+          <button onClick={this.onClickHandler} type="button" name="6">6</button>
+          <button onClick={this.onClickHandler} type="button" id="less">-;</button>
+          <button onClick={this.onClickHandler} type="button" name="1">1</button>
```

Comment:
- [ ] Great job with your project 👏 . Unfortunately, when I click the subtract button it doesn't work. Kindly fix this 🙏